fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -2130,6 +2130,7 @@ private static List<Field> normalizeFields(@Nullable final List<Field> fields) {
                 }
                 if (field.fieldIndexOptional.isPresent() && !fieldIndexesSeen.add(field.getFieldIndex())) {
                     override = true;
+                    break;
```

Comment:
This is a small optimization so that once we find a duplicated field index, we stop looking. The `fieldIndexesSeen` set is not referenced after this loop.